File: lib/src/models/post_response.dart
import 'package:json_annotation/json_annotation.dart';

import 'ignore_response.dart';

part 'post_response.g.dart';

@JsonSerializable()
class PostResponse{
	
	PostResponse();

	factory PostResponse.fromJson(Map<String, dynamic> json) => _$PostResponseFromJson(json);

	@JsonKey(name:'lastReplyTimestamp')
	String lastReplyTimestamp;
	@JsonKey(name:'IsPinned')
	bool isPinned;
	@JsonKey(name:'urlMediaType')
	int urlMediaType;
	@JsonKey(name:'thumbnail')
	String thumbnail;
	@JsonKey(name:'popularity')
	int popularity;
	@JsonKey(name:'isActive')
	bool isActive;
	@JsonKey(name:'isAnnouncement')
	bool isAnnouncement;
	@JsonKey(name:'userRating')
	int userRating;
	@JsonKey(name:'userHasRated')
	bool userHasRated;
	@JsonKey(name:'userHasMutedPost')
	bool userHasMutedPost;
	@JsonKey(name:'latestReplyPostId')
	String latestReplyPostId;
	@JsonKey(name:'latestReplyAuthorId')
	String latestReplyAuthorId;
	@JsonKey(name:'ignoreStatus')
	IgnoreResponse ignoreStatus;
	@JsonKey(name:'locale')
	String locale;

	
	
	Map<String, dynamic> toJson() => _$PostResponseToJson(this);
}

File: pkg/models/PostResponse.go
package bungieapigo

import "time"

type PostResponse struct {
	LastReplyTimestamp  time.Time           `json:"lastReplyTimestamp"`
	IsPinned            bool                `json:"IsPinned"`
	UrlMediaType        ForumMediaType      `json:"urlMediaType"`
	Thumbnail           string              `json:"thumbnail"`
	Popularity          ForumPostPopularity `json:"popularity"`
	IsActive            bool                `json:"isActive"`
	IsAnnouncement      bool                `json:"isAnnouncement"`
	UserRating          int                 `json:"userRating"`
	UserHasRated        bool                `json:"userHasRated"`
	UserHasMutedPost    bool                `json:"userHasMutedPost"`
	LatestReplyPostId   int64               `json:"latestReplyPostId,string"`
	LatestReplyAuthorId int64               `json:"latestReplyAuthorId,string"`
	IgnoreStatus        IgnoreResponse      `json:"ignoreStatus"`
	Locale              string              `json:"locale"`
}
